Qt5.15.2 mingw 链接库没有 d后缀

### Xmake 版本

v2.7.9+HEAD.c87922676

### 操作系统版本和架构

Window10

### 描述问题

在 Qt 5.15.2 中 mingw环境下 没有 d后缀的库
但是 xmake 给加上了 d 导致编译失败

不知道 是不是所有mingw版本都没有 d后缀

### 期待的结果

可以正常编译

### 工程配置

```
target("test")
    add_rules("qt.widgetapp")
    add_files("main.cpp", {rules = "qt.moc"})

    add_frameworks("QtCore", "QtWidgets")
target_end()
```
```
#include <QPushButton>
#include <QApplication>

int main(int argc, char *argv[])
{
    QApplication a(argc, argv);
    QPushButton w;
    w.show();
    return a.exec();
}
```


### 附加信息和错误日志

```
xmake f -p mingw -a x86_64 -m debug --mingw=D:/Qt/Tools/mingw810_64 --qt=D:/Qt/5.15.2/mingw81_64
checking for mingw directory ... D:/Qt/Tools/mingw810_64
checking for Qt SDK directory ... D:/Qt/5.15.2/mingw81_64
checking for Qt SDK version ... 5.15.2
PS C:\Users\zhuizhu\Desktop\mingwbug> xmake -vD      
checking for x86_64-w64-mingw32-g++ ... D:\Qt\Tools\mingw810_64\bin\x86_64-w64-mingw32-g++
checking for the c++ compiler (cxx) ... x86_64-w64-mingw32-g++
[ 20%]: compiling.qt.moc main.cpp
D:\Qt\5.15.2\mingw81_64\bin\moc.exe -DQT_QML_DEBUG -DQT_DEPRECATED_WARNINGS -DQT_GUI_LIB -DQT_WIDGETS_LIB -DQT_CORE_LIB -IC:/Users/zhuizhu/Desktop/mingwbug/build/.gens/test_core/mingw/x86_64/debug/rules/qt/ui -ID:/Qt/5.15.2/mingw81_64/include/QtGui -ID:/Qt/5.15.2/mingw81_64/include/QtGui/5.15.2 -ID:/Qt/5.15.2/mingw81_64/include/QtGui/5.15.2/QtGui -ID:/Qt/5.15.2/mingw81_64/include/QtWidgets -ID:/Qt/5.15.2/mingw81_64/include/QtWidgets/5.15.2 -ID:/Qt/5.15.2/mingw81_64/include/QtWidgets/5.15.2/QtWidgets -ID:/Qt/5.15.2/mingw81_64/include/QtCore -ID:/Qt/5.15.2/mingw81_64/include/QtCore/5.15.2 -ID:/Qt/5.15.2/mingw81_64/include/QtCore/5.15.2/QtCore -ID:/Qt/5.15.2/mingw81_64/include -ID:/Qt/5.15.2/mingw81_64/mkspecs/win32-g++ main.cpp -o build\.gens\test_core\mingw\x86_64\debug\main.moc
main.cpp:0: Note: No relevant classes found. No output generated.
[ 40%]: cache compiling.debug main.cpp
D:\Qt\Tools\mingw810_64\bin\x86_64-w64-mingw32-g++ -c -m64 -std=c++11 -IC:\Users\zhuizhu\Desktop\mingwbug\build\.gens\test_core\mingw\x86_64\debug\rules\qt\ui -Ibuild\.gens\test_core\mingw\x86_64\debug -DQT_QML_DEBUG -DQT_DEPRECATED_WARNINGS -DQT_GUI_LIB -DQT_WIDGETS_LIB -DQT_CORE_LIB -isystem D:\Qt\5.15.2\mingw81_64\include\QtGui -isystem D:\Qt\5.15.2\mingw81_64\include\QtGui\5.15.2 -isystem D:\Qt\5.15.2\mingw81_64\include\QtGui\5.15.2\QtGui -isystem D:\Qt\5.15.2\mingw81_64\include\QtWidgets -isystem D:\Qt\5.15.2\mingw81_64\include\QtWidgets\5.15.2 -isystem D:\Qt\5.15.2\mingw81_64\include\QtWidgets\5.15.2\QtWidgets -isystem D:\Qt\5.15.2\mingw81_64\include\QtCore -isystem D:\Qt\5.15.2\mingw81_64\include\QtCore\5.15.2 -isystem D:\Qt\5.15.2\mingw81_64\include\QtCore\5.15.2\QtCore -isystem D:\Qt\5.15.2\mingw81_64\include -isystem D:\Qt\5.15.2\mingw81_64\mkspecs\win32-g++ -o build\.objs\test_core\mingw\x86_64\debug\main.cpp.obj main.cpp
checking for D:\Qt\Tools\mingw810_64\bin\x86_64-w64-mingw32-g++ ... ok
checking for flags (-MMD -MF) ... ok
> x86_64-w64-mingw32-g++ "-MMD" "-MF" "C:\Windows\TEMP\.xmake\230628\_18AFDF68C9494D5080A55FABC1C1FDA0" "-m64"
checking for flags (-fdiagnostics-color=always) ... ok
> x86_64-w64-mingw32-g++ "-fdiagnostics-color=always" "-m64"
checking for x86_64-w64-mingw32-g++ ... D:\Qt\Tools\mingw810_64\bin\x86_64-w64-mingw32-g++
checking for the linker (ld) ... x86_64-w64-mingw32-g++
[ 60%]: linking.debug test_core.exe
D:\Qt\Tools\mingw810_64\bin\x86_64-w64-mingw32-g++ -o build\mingw\x86_64\debug\test_core.exe build\.objs\test_core\mingw\x86_64\debug\main.cpp.obj -m64 -LD:\Qt\5.15.2\mingw81_64\lib -lQt5Guid -lQt5Widgetsd -lQt5Cored -lmingw32 -lws2_32 -lgdi32 -lole32 -ladvapi32 -lshell32 -luser32 -liphlpapi -mwindows
D:/Qt/Tools/mingw810_64/bin/../lib/gcc/x86_64-w64-mingw32/8.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lQt5Guid
D:/Qt/Tools/mingw810_64/bin/../lib/gcc/x86_64-w64-mingw32/8.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lQt5Widgetsd
D:/Qt/Tools/mingw810_64/bin/../lib/gcc/x86_64-w64-mingw32/8.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lQt5Cored
```

--- a/xmake/rules/qt/load.lua
+++ b/xmake/rules/qt/load.lua
@@ -32,7 +32,11 @@ function _link(target, linkdirs, framework, qt_sdkver)
         if target:is_plat("windows") then
             debug_suffix = "d"
         elseif target:is_plat("mingw") then
-            debug_suffix = "d"
+            if qt_sdkver:ge("5.15.2") then
+                debug_suffix = ""
+            else
+                debug_suffix = "d"
+            end
         elseif target:is_plat("android") or target:is_plat("linux") then
             debug_suffix = ""
         end

